### PR TITLE
CLOUDP-148830: Added CRD versions logic

### PR DESCRIPTION
**Pull request by igor-karpukhin**
### All Submissions:

Added logic to handle versions of operator's custom resources

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: pkg/api/v1/atlasbackuppolicy_types.go [hunk 1]
```diff
@@ -47,6 +49,15 @@ type AtlasBackupPolicy struct {
 	Status AtlasBackupPolicyStatus `json:"status,omitempty"`
 }
 
+func (in *AtlasBackupPolicy) GetStatus() status.Status {
+	// TODO: Use status field
```
Comment (fabritsius) [inline pkg/api/v1/atlasbackuppolicy_types.go:53]: Is it ok to leave like this?

**Review by fabritsius — APPROVED** [review]
Looks Great 👍 
Just had one small question :)

**Review by Sugar-pack — APPROVED** [review]
LGTM

File: pkg/api/v1/atlasbackuppolicy_types.go (hunk 1 above)
Comment (igor-karpukhin) [reply pkg/api/v1/atlasbackuppolicy_types.go]: Removed

**Pull request merged**